### PR TITLE
Overhaul

### PR DESCRIPTION
**Pull request by yurrriq**
See commit messages for details.

I'm happy to remove or change parts of this as necessary.

**Edit**: This PR now includes #6 and #7.

### CODE REVIEW

**Comment by yurrriq** [comment]
Build failure is some kind of Coveralls issue.

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,5 @@
 .rebar3
 rebar3.crashdump
-_*
```
Comment (deadtrickster) [inline .gitignore:3]: Why?? :laughing:

**Comment by deadtrickster** [comment]
Regarding this ones: 

 @yurrriq	[smerl.erl] Add code server support  …	Verified		7bde1c3
 @yurrriq	[smerl.erl] Add options support (src_dir)  …	Verified		7c963d3
 @yurrriq	[smerl.erl,scratch.erl] Rearrange/refactor code  …	Verified		7f1cdae


are they documented somewhere? If not, maybe we can split them up and add docs in separate PR?

**Comment by yurrriq** [comment]
The trouble is I did the rearranging and fix for #2 ion the options-support branch which depended on the other two PRs.. they're not yet documented, but the API hasn't changed. I did change `compile/2` s.t. `undefined` is no longer valid for `Options`.

**Comment by deadtrickster** [comment]
Awesome job. :tada:

**Pull request merged**